gateway/ln-gateway/src/lib.rs

```diff
@@ -64,6 +65,7 @@ impl LnGateway {
         let ln_gw = Self {
             config,
             actors: Mutex::new(HashMap::new()),
+            channels: Mutex::new(HashSet::new()),
```

Comment:
This field needs to be populated on restarts/when loading actors from config imo. Maybe having a separate channels struct isn't ideal and we should just query the actors for their IDs and find a new one?

Reply:
> This field needs to be populated on restarts/when loading actors from config imo.

True. and since I persist a previously assigned channel id on the config, naive new assignments would conflict.

> Maybe having a separate channels struct isn't ideal and we should just query the actors for their IDs and find a new one?

I'm keen on keeping new channel id  assignment to scale sub-linear with the count of actors. worst case should be O(n). Will reconsider my approach here

Reply:
Okay I think I got it with the latest iteration. Check it out